src/components/ProtectedContent/ProtectedContent.tsx

```diff
@@ -1,63 +1,92 @@
 import { ErrorPage } from 'components/ErrorPage/ErrorPage';
+import { KeycloakStatusPage } from 'components/KeycloakStatusPage/KeycloakStatusPage';
 
 import { AUTH_ROLE, keycloakService } from 'services/keycloakService';
+import { uiLogger } from 'services/uiLogger';
 
 import styles from './ProtectedContent.module.css';
 
 export enum PROTECTED_TYPE {
-  Route = 'route',
-  Component = 'component',
+  Route,
+  ComponentDisabled,
+  ComponentHidden,
 }
 
 interface IProtectedContentProps {
   type: PROTECTED_TYPE;
   role?: AUTH_ROLE;
   title?: string;
-  hide?: boolean;
-  disable?: boolean;
 }
 
+const HiddenContent = () => null;
+
+interface IDisabledContentProps {
+  reason: string;
+}
+const DisabledContent = ({ children, reason }: React.PropsWithChildren<IDisabledContentProps>) => (
+  <div title={reason}>
+    <div className={styles['disabled-content']}>{children}</div>
+  </div>
+);
+
+/**
+ * Internal component.
+ *
+ * Only {@link ProtectedComponent} and {@link ProtectedRoute} are assumed to use this component directly and they should be used instead.
```

Comment:
`{@link ProtectedComponent} and {@link ProtectedRoute} use this component. They are meant to be used directly instead of this component.`